### PR TITLE
Update DG with sort feature and UML diagrams

### PR DESCRIPTION
**Pull request by ChryslineLim**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -494,6 +494,52 @@ The below UML diagram represents the overall structure of the predicates for `Bu
...
+on entering a command. As such, the last-shown list changes depending on the last command entered. To keep track of this,
+we used a flag which would be updated everytime a `filter` or `sort` command was used. All `Command`s that referred to an
+entry on the displayed list were adapted to take relative indices from the last-shown list indicated by the flag. The `Model`
+component also needed access to the `UI` component's `BuyerPanelList` and `PropertyPanelList` in order to display
+the corresponding `FilteredList` or `SortedList` based on changes to the flag. To reduce coupling, we would have had to
+apply the Observer pattern.
+
+We decided against this as the complicated design made it more bug-prone. In addition, it did not allow for
+stacking of `filter` and `sort` functions, that is, a user is unable to filter on top of a sorted list or vice versa as the
+`FilteredList` and `SortedList` are independent and separate from each other.
```
Comment (chngchngchng) [inline docs/DeveloperGuide.md:537]: Perhaps this portion of the writeup could be included under design considerations, following the structure of other parts of the dg above!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -494,6 +494,52 @@ The below UML diagram represents the overall structure of the predicates for `Bu
...
+![PropertyComparatorsClassDiagram](images/SortPropComparatorsClassDiagram.png)
+
+#### Design Considerations
+Similar to the `FilteredList` abstraction provided by JavaFX, we considered using a `SortedList` to present the list in a
+sorted version without modifying the underlying data structure `ObservableArrayList`. This is
+to preserve the chronological order in which users enter the entries so that it can still be displayed with the `list` command.
+
+However, this meant that we needed to have both `FilteredList` and `SortedList` stored and vary which one is displayed to users
+on entering a command. As such, the last-shown list changes depending on the last command entered. To keep track of this,
+we used a flag which would be updated everytime a `filter` or `sort` command was used. All `Command`s that referred to an
```
Comment (chngchngchng) [inline docs/DeveloperGuide.md:529]: Maybe we should explain this as an "overall" thing for both filter and sort?

**Review by chngchngchng — APPROVED** [review]
LGTM! Good work creating the UML diagrams :)

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ChryslineLim) [reply docs/DeveloperGuide.md:537]: What do you mean haha it's under the design considerations section! Or did you mean something else?

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (ChryslineLim) [reply docs/DeveloperGuide.md:529]: Like after the 2 sections?

**Pull request merged**